src/Apia/Defaults.hs

```diff
@@ -0,0 +1,359 @@
+where
+
+import           Apia.Prelude
+
+import           Agda.Utils.Trie       as Trie
+import           Apia.Common           (Lang (SMT2, TPTP))
+import           Apia.Options          (Options (..), verboseOpt)
+import           Control.DeepSeq       (NFData (rnf))
+import qualified Data.HashMap.Strict   as HashMap
+import qualified Data.Text             as T
+import           Data.Yaml             (FromJSON (parseJSON), Object,
+                                        parseMaybe)
+import qualified Data.Yaml.Include     as YamlInclude
+import           Paths_apia            (getDataFileName)
+import           Prelude               hiding (lookup)
+import           System.Directory      (copyFile, doesFileExist,
+                                        getCurrentDirectory, getHomeDirectory)
+import           System.FilePath.Posix ((</>))
+
+-- | Default options use by the program.
+defaultOptions ∷ Options
+defaultOptions = Options
+  { optATP                             = []
+  , optCheck                           = False
+  , optDumpTypes                       = False
+  , optFnConstant                      = False
+  , optHelp                            = False
+  , optIncludePath                     = []
+  , optInputFile                       = Nothing
+  , optLang                            = TPTP
+  , optNoInternalEquality              = False
+  , optNoPredicateConstants            = False
+  , optOnlyFiles                       = False
+  , optOutputDir                       = "/tmp"
+  , optSchematicFunctions              = False
+  , optSchematicPropositionalFunctions = False
+  , optSchematicPropositionalSymbols   = False
+  , optSnapshotDir                     = "snapshot"
+  , optSnapshotNoError                 = False
+  , optSnapshotTest                    = False
+  , optTime                            = 240
+  , optUnprovenNoError                 = False
+  , optVerbose                         = Trie.singleton [] 1
+  , optVersion                         = False
+  , optWithCVC4                        = "cvc4"
+  , optWithE                           = "eprover"
+  , optWithEquinox                     = "equinox"
+  , optWithIleanCoP                    = "ileancop.sh"
+  , optWithMetis                       = "metis"
+  , optWithSPASS                       = "SPASS"
+  , optWithtptp4X                      = "tptp4X"
+  , optWithVampire                     = "vampire_lin64"
+  , optWithZ3                          = "z3"
+}
+
+type Key = T.Text
+
+data Config = Config Object
+  deriving (Eq, Show)
+
+instance NFData Config where
+  rnf (Config o) = rnf o `seq` ()
+
+look ∷ FromJSON b => Key → Config → Maybe b
+look k (Config o) = HashMap.lookup k o >>= parseMaybe parseJSON
+
+setVals ∷ Options → Config → Options
+setVals = setATP
+
+setATP ∷ Options → Config → Options
+setATP opts config = setCheck newopts config
+  where
+    newopts ∷ Options
+    newopts = case look "atp" config of
+      (Just  val) → opts { optATP = val }
```

Comment:
Better `Just val`.  HLint should report it. Please create an issue on HLint CC'ing me.